### PR TITLE
Features/97 blocking post wait

### PR DESCRIPTION
**Pull request by mitch-lbw**

### CODE REVIEW

File: src/main/java/com/kryptokrauts/aeternity/sdk/service/transaction/impl/TransactionServiceImpl.java [hunk 1]
```diff
@@ -173,4 +187,46 @@ private Tx createGeneratedTxObject(String signedTx) {
...
+      while (currentBlockHeight == -1
+          && elapsedTrials < this.config.getNumTrialsToWaitForTxMined()) {
+        minedTransaction = this.externalApi.rxGetTransactionByHash(transactionHash).blockingGet();
+        if (minedTransaction.getBlockHeight().intValue() > 1) {
+          _logger.debug("Transaction is mined - " + minedTransaction.toString());
+          currentBlockHeight = minedTransaction.getBlockHeight().intValue();
+        } else {
+          _logger.info(
+              String.format(
+                  "Transaction not mined yet, checking again in %s seconds (trial %s of %s)",
```
Comment (marc0olo) [inline src/main/java/com/kryptokrauts/aeternity/sdk/service/transaction/impl/TransactionServiceImpl.java]: I think we should convert the seconds value to a double here just to make sure we always display the correct value

**Comment by marc0olo** [comment]
for some reason the build fails:
```
│  ├─ callContractOnLocalNodeTest ✘ Multiple Failures (2 failures)
   │  │        	Not equals : th_L9ed94RnrjXiQm8Uvp4nUw2fZe4vFwCxi3J3dcfS1c7dr8mP8 != th_2AnuA1LcJFwiNChKkgDmkzZPt7a4uerngKpwW7HzUrGn5RZRd2
   │  │        	Test failed
```

**Comment by mitch-lbw** [comment]
The cause for the failing test is this line
`context.assertEquals(result.getTxHash(),
					this.aeternityServiceNative.transactions.computeTxHash(
							createCallContractModel(getNextBaseKeypairNonce(),
									callData, null)));`
At this point the assert absolutely makes no sense, because after succesful postTx the nonce is one value higher - @marc0olo any idea of the initial purpose?

File: src/main/java/com/kryptokrauts/aeternity/sdk/service/transaction/impl/TransactionServiceImpl.java (hunk 1 above)
Comment (mitch-lbw) [reply src/main/java/com/kryptokrauts/aeternity/sdk/service/transaction/impl/TransactionServiceImpl.java]: Print is adapted to display exact seconds value

**Comment by marc0olo** [comment]
> The cause for the failing test is this line
> `context.assertEquals(result.getTxHash(), this.aeternityServiceNative.transactions.computeTxHash( createCallContractModel(getNextBaseKeypairNonce(), callData, null)));`
> At this point the assert absolutely makes no sense, because after succesful postTx the nonce is one value higher - @marc0olo any idea of the initial purpose?

we wanted to verify that the txHash is being calculated correctly here. as you already mentioned there was an issue by using the next nonce which in this case is being requested from the node api. in the past we had another mining configuration and therefore we didn't notice the problem. I changed this so that `computeTxHash` is using the same nonce.

**Pull request merged**